scivision/catalog/catalog.py

```diff
@@ -22,12 +23,22 @@ def __init__(self):
     def URI(self):
         return self._uri
 
-    def query(self, query: dict) -> list:
+    def query(self, query: Dict[str, str]) -> list:
         query = _validate_query(query)
-        return self._query(query)
+        return self._query(query) if query else []
 
     @abc.abstractmethod
-    def _query(self, query: dict) -> list:
+    def _query(self, query: Dict[str, str]) -> list:
+        raise NotImplementedError
+
+    @property
+    @abc.abstractmethod
+    def keys(self) -> List[str]:
+        raise NotImplementedError
+
+    @property
```

Comment:
```suggestion
    @property
```
oops - this shouldn't be a property